server/sdserver/server.go

```diff
@@ -44,14 +44,26 @@ var Set = wire.NewSet(
 )
 
 // NewExporter returns a new OpenCensus Stackdriver exporter.
-func NewExporter(id gcp.ProjectID, ts gcp.TokenSource, mr monitoredresource.Interface) (*stackdriver.Exporter, error) {
+//
+// The second return value is a Wire cleanup function that calls Flush
+// on the exporter.
+func NewExporter(id gcp.ProjectID, ts gcp.TokenSource, mr monitoredresource.Interface) (*stackdriver.Exporter, func(), error) {
 	tokOpt := option.WithTokenSource(oauth2.TokenSource(ts))
-	return stackdriver.NewExporter(stackdriver.Options{
+	exp, err := stackdriver.NewExporter(stackdriver.Options{
 		ProjectID:               string(id),
 		MonitoringClientOptions: []option.ClientOption{tokOpt},
 		TraceClientOptions:      []option.ClientOption{tokOpt},
 		MonitoredResource:       mr,
 	})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	cleanup := func() {
+		exp.Flush()
+	}
+
+	return exp, cleanup, err
```

Comment:
nit: it seems you can just `return exp, exp.Flush(), err`

Reply:
You meant
```go
return exp, func() { exp.Flush() }, err
```
?

Reply:
My bad, either like you said or just `return exp, exp.Flush, err`

Reply:
Thanks, I ended up with `func() { exp.Flush() }` since its style would be more consistent with `xrayserver`